lib/chat_channel_fetcher.rb

```diff
@@ -26,7 +26,15 @@ def self.secured_public_channels(guardian, memberships, scope_with_membership: t
         .where(user_chat_channel_memberships: { user_id: guardian.user.id, following: true })
     end
 
-    filter_public_channels(channels, memberships, guardian)
+    channels = filter_public_channels(channels, memberships, guardian).to_a
+
+    preload_fields = Category.instance_variable_get(:@custom_field_types).keys
```

Comment:
There's not way to access `custom_field_types` more directly AFAICT. I could add something in core, but we could still not use it right now as it would force people to update right away to use chat.

But let me know if you think I should add it to core so we can use it in the future.

Reply:
Yeah, a proper api would be good. But rather than exposing `custom_field_types` maybe it would be better to change `#preload_custom_fields` to allow preloading all custom fields. Because that's the use case, right?